### PR TITLE
Generate process metrics with semconv yaml

### PR DESCRIPTION
**Pull request by braydonk**

### CODE REVIEW

**Comment by braydonk** [comment]
Hi @joaopgrassi, I decided to open my changes as a draft PR so you could check my work. This is converting the metrics from `process-metrics.md` into yaml that `semconvgen` can use. I also included the example of one generation in the markdown file. I would appreciate if you could take a look and make sure I'm on the right track before I finish off the work and open a properly-formed PR.

File: semantic_conventions/metrics/process-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
+groups:
+  - id: process.cpu
```
Comment (joaopgrassi) [inline semantic_conventions/metrics/process-metrics.yaml:2]:
```suggestion
  - id: attributes.process.cpu
```

I started adding `attributes.` to `attribute_group` to easily spot that this is not a span/metric/log convention but just rather a "bag" of attributes to be shared.

**Review by joaopgrassi — COMMENTED** [review]
@braydonk looks pretty good to me! I think all is in the right track :)

File: semantic_conventions/metrics/process-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    brief: "Number of file descriptors in use by the process."
+    instrument: updowncounter
+    unit: "{count}"
+    attributes: []
+
+  - id: metric.process.context_switches
+    type: metric
+    metric_name: process.context_switches
+    brief: "Number of times the process has been context switched."
+    instrument: updowncounter
```
[the remark this answers is not in the document]
Comment (braydonk) [reply semantic_conventions/metrics/process-metrics.yaml:109]: Nope we want to keep a counter, that was a copy paste mishap :smile: thanks

**Review by jsuereth — APPROVED** [review]
Looks like you have a minor YAMLlint error

**Review by mx-psi — COMMENTED** [review]
I have a similar comment to the one I left in https://github.com/open-telemetry/semantic-conventions/pull/89#pullrequestreview-1571542862; this PR moves the metrics to YAML and also renames the attributes. If doing each of those changes separately is hard/undesirable, I think we should at least describe this clearly in the PR title, PR description and a changelog entry indicating what changes #51 entails for this set of metrics

**Comment by braydonk** [comment]
@mx-psi I'm kind of confused, which part of this PR needs to be separated? I don't think I intentionally renamed any attributes; I tried to copy them directly from the existing document.

**Comment by braydonk** [comment]
Thanks @joaopgrassi I'll get that fixed soon.

**Comment by braydonk** [comment]
I've opened #330 instead. The commit history on this one was really messy with attempts to keep it up to date with main branch, so I started a new clean PR.

**Pull request closed without merging**